components-core/src/test/java/com/adyen/checkout/components/core/internal/analytics/data/DefaultAnalyticsRepositoryTest.kt

```diff
@@ -128,4 +151,48 @@ internal class DefaultAnalyticsRepositoryTest(
             verify(localErrorDataStore, never()).removeEvents(any())
         }
     }
+
+    companion object {
+
+        @OptIn(DirectAnalyticsEventCreation::class)
+        @JvmStatic
+        fun sendEventsSource() = listOf(
```

Comment:
I think we should cover the case where all three of the lists are empty and we don't send any event. What do you think?

Reply:
We already have a test called `there are no events stored, then sending is canceled` for it at `DefaultAnalyticsRepositoryTest:90`.